### PR TITLE
atmos: init at 1.4.28

### PR DESCRIPTION
**Pull request by nitrocode**
###### Description of changes

- Add [atmos](https://github.com/cloudposse/atmos)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/atmos/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  vendorSha256 = "sha256-bepv8Ns8vcpLpmO7fsRgGijB9B6kQ5HhxUhIuEpGtOY=";
+
+  ldflags = [ "-X 'github.com/cloudposse/atmos/cmd.Version=${version}'" ];
+
+  doCheck = false;
+
+  doInstallCheck = true;
+  installCheckPhase = ''
```
Comment (onny) [inline pkgs/applications/networking/cluster/atmos/default.nix:21]: Is it possible to use a more standardized way to check this? Something like
``` 
 passthru.tests.version = testers.testVersion {
   package = opensupaplex;
   command = "opensupaplex --help";
   version = "v${version}";
 };
```

File: pkgs/applications/networking/cluster/atmos/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-AqW//+x1tKzm5dJirXCLFuDBTk3dHHw7c6vwsVsI37U";
+  };
+
+  vendorSha256 = "sha256-bepv8Ns8vcpLpmO7fsRgGijB9B6kQ5HhxUhIuEpGtOY=";
+
+  ldflags = [ "-X 'github.com/cloudposse/atmos/cmd.Version=${version}'" ];
+
+  doCheck = false;
```
Comment (onny) [inline pkgs/applications/networking/cluster/atmos/default.nix:18]: Doesn't include any tests? If it does please specify why you have to disable it

File: pkgs/applications/networking/cluster/atmos/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "cloudposse";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-AqW//+x1tKzm5dJirXCLFuDBTk3dHHw7c6vwsVsI37U";
+  };
+
+  vendorSha256 = "sha256-bepv8Ns8vcpLpmO7fsRgGijB9B6kQ5HhxUhIuEpGtOY=";
+
+  ldflags = [ "-X 'github.com/cloudposse/atmos/cmd.Version=${version}'" ];
```
Comment (06kellyjac) [inline pkgs/applications/networking/cluster/atmos/default.nix:16]:
```suggestion
  ldflags = [ "-X github.com/cloudposse/atmos/cmd.Version=v${version}" ];
```
needs a v prefix

```
/ # ./atmos_1.4.28_linux_amd64 version
v1.4.28
```

You can put single quotes around it but it's not strictly necessary unless there are spaces

File: pkgs/applications/networking/cluster/atmos/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    $out/bin/atmos version | grep "${version}"
+    runHook postInstallCheck
+  '';
+
+  meta = with lib; {
+    homepage = "https://atmos.tools";
+    changelog = "https://github.com/cloudposse/atmos/releases/tag/v${version}";
+    description = "Universal Tool for DevOps and Cloud Automation (works with terraform, helm, helmfile, etc)";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ jk ];
```
Comment (06kellyjac) [inline pkgs/applications/networking/cluster/atmos/default.nix:32]:
```suggestion
    maintainers = with maintainers; [ ];
```

Please don't sign me up as maintainer :joy:

Looks like you'll need to add yourself to the maintainers list. please do this in a separate commit as explained in CONTRIBUTING.md

File: pkgs/applications/networking/cluster/atmos/default.nix (hunk 1 above)
Comment (06kellyjac) [reply pkgs/applications/networking/cluster/atmos/default.nix:21]: I do it via `installCheckPhase` much more often and there's been a lot of discussions around the positives and negatives (I can't find any links ATM)

It's not really standardized, just completely random

```
λ rg "version \| " --files-with-matches | grep ".nix" | grep -v "/tests/" | wc -l
73
λ rg "testers\.testVersion \{" --files-with-matches | wc -l
131
```

my main reasons for not doing the passthru is 1: I need to remember to build `xyz.passthru.tests.version`, 2: I don't like having to import the package into itself to provide it to the test passthru because it's hideous

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
Please don't add me as a maintainer :)

File: pkgs/applications/networking/cluster/atmos/default.nix (hunk 1 above)
Comment (nitrocode) [reply pkgs/applications/networking/cluster/atmos/default.nix:21]: Ok, I will leave this as-is then.

File: pkgs/applications/networking/cluster/atmos/default.nix (hunk 2 above)
Comment (nitrocode) [reply pkgs/applications/networking/cluster/atmos/default.nix]: Fixed!

File: pkgs/applications/networking/cluster/atmos/default.nix (hunk 3 above)
Comment (nitrocode) [reply pkgs/applications/networking/cluster/atmos/default.nix]: Fixed!

File: pkgs/applications/networking/cluster/atmos/default.nix (hunk 4 above)
Comment (nitrocode) [reply pkgs/applications/networking/cluster/atmos/default.nix]: Fixed! I will add a new maintainer in a separate PR then.

File: pkgs/applications/networking/cluster/atmos/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "cloudposse";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-AqW//+x1tKzm5dJirXCLFuDBTk3dHHw7c6vwsVsI37U";
+  };
+
+  vendorSha256 = "sha256-bepv8Ns8vcpLpmO7fsRgGijB9B6kQ5HhxUhIuEpGtOY=";
+
+  ldflags = [ "-X github.com/cloudposse/atmos/cmd.Version=v${version}" ];
```
Comment (azahi) [inline pkgs/applications/networking/cluster/atmos/default.nix:16]: This will strip the executable.
```suggestion
  ldflags = [ "-s" "-w" "-X github.com/cloudposse/atmos/cmd.Version=v${version}" ];
```

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
You cant add a package without a maintainer
Please add yourself as a maintainer or try find one

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/atmos/default.nix (hunk 5 above)
Comment (nitrocode) [reply pkgs/applications/networking/cluster/atmos/default.nix]: This is different than the upstream Makefile but I have added this change

https://github.com/cloudposse/atmos/blob/994003239ba18aca9ac688f511dd0f13228f9529/Makefile#L21

**Comment by nitrocode** [comment]
@onny (cc: @azahi @06kellyjac) may I set `doCheck` back to `false` ? I'm seeing `Unexpected errors` in the output and I'm unsure on how to troubleshoot.

**Comment by azahi** [comment]
> @onny (cc: @azahi @06kellyjac) may I set `doCheck` back to `false` ? I'm seeing `Unexpected errors` in the output and I'm unsure on how to troubleshoot.

This works for me:
```nix
  preCheck = ''
    # Remove tests that depend on a network connection.
    rm -rf pkg/vender/component_vendor_test.go
  '';
```

**Comment by 06kellyjac** [comment]
A way to debug is to set the timeout to faster so it fails quicker & you see which tests are using networking

**Comment by onny** [comment]
Thank you!

**Comment by nitrocode** [comment]
@onny when will I be able to run `nix-shell -p atmos` for example? Does it take a while to get into the 22.05 repo channel?

**Comment by onny** [comment]
> @onny when will I be able to run `nix-shell -p atmos` for example? Does it take a while to get into the 22.05 repo channel?

Currently it is only available in unstable channel. We could backport it or wait for 21.11 release.

But you can run it from unstable anytime
```
nix run github:NixOS/nixpkgs#atmos
```

**Comment by nitrocode** [comment]
Thanks onny! A few more questions for you if you don't mind.

- What is the process to backport it? Is that documented somewhere?
- When does 21.11 release come out? How often are release cycles?

**Comment by onny** [comment]
Not sure about backporting but there should be documentation about it somewhere.
Stable releases will be twice a year in November and May
